contracts/examples/IdentityExample.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.16;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "../interfaces/IState.sol";
+import "../lib/ClaimBuilder.sol";
+import "../lib/IdentityLib.sol";
+
+// /**
+//  * @dev Contract managing onchain identity
+//  */
+contract IdentityExample is OwnableUpgradeable {
+    using IdentityLib for IdentityLib.Trees;
+    using IdentityLib for IdentityLib.IdentityData;
+
+    uint256 public constant IDENTITY_MAX_SMT_DEPTH = 40;
+
+    IdentityLib.IdentityData public identity;
+    IdentityLib.Trees internal trees;
+    IdentityLib.LastTrees public lastTrees;
```

Comment:
Why we have `using ... for ...` for `IdentityLib.Trees` and `IdentityLib.IdentityData`. But have not for `IdentityLib.LastTrees`?

Reply:
LastTrees data doesn't have any extension methods (self).
Others two have.